hail/src/main/scala/is/hail/shuffler/LSM.scala

```diff
@@ -122,8 +139,128 @@ class LSM (
     }
   }
 
-  val store = new StoreBuilder[Long, Long](new File(path),
+  private[this] val store = new StoreBuilder[Long, Long](new File(path),
     new RegionValueSerializer(keyDec _, keyEnc _),
     new RegionValueSerializer(dec _, enc _)
-  ).setComparator(ord).build()
+  ).setComparator(keyOrd).build()
+  private[this] val rnd = new Random()
+  private[this] var processed = 0L
+  private[this] var least = -1L
+  private[this] var greatest = -1L
+  private[this] var samples = new Array[Long](LSM.nKeySamples - 2)
+  private[this] var sorted = false
+  private[this] var samplesEnd = 0
+
+  private[this] def maybeSample(k: Long): Unit = {
+    if      (processed == 0)
+      assert(least == -1 && greatest == -1)
+    else if (processed == 1)
+      assert(least != -1 && greatest == -1)
+    else if (processed >= 2)
+      assert(keyOrd.compare(least, greatest) < 0)
+    assert(samplesEnd <= samples.length)
```

Comment:
Are these assertions redundant based on the ones later in the function?

Reply:
Either this, or are the later assertions redundant?

Reply:
Sure are, I've removed the later ones.